rasa/server.py

```diff
@@ -1277,36 +1474,30 @@ def _validate_json_training_payload(rjs: Dict):
         )
 
 
-def _training_payload_from_yaml(request: Request) -> Dict[Text, Union[Text, bool]]:
+def _training_payload_from_yaml(
+    request: Request, temp_dir: Path
+) -> Dict[Text, Union[Text, bool]]:
     logger.debug("Extracting YAML training data from request body.")
 
     decoded = request.body.decode(rasa.shared.utils.io.DEFAULT_ENCODING)
     _validate_yaml_training_payload(decoded)
 
-    temp_dir = tempfile.mkdtemp()
-    training_data = Path(temp_dir) / "data.yml"
+    training_data = temp_dir / "data.yml"
     rasa.shared.utils.io.write_text_file(decoded, training_data)
 
-    model_output_directory = _model_output_directory(
-        request.args.get("save_to_default_model_directory", True)
-    )
+    model_output_directory = str(temp_dir)
+    if request.args.get("save_to_default_model_directory", True):
```

Comment:
In my opinion request args should all be extracted at the top level.

Reply:
you mean right in the request handler instead of submethod?